pkg/kubelet/kubelet_pods.go

```diff
@@ -886,12 +887,17 @@ func (kl *Kubelet) getPullSecretsForPod(pod *v1.Pod) []v1.Secret {
 		secret, err := kl.secretManager.GetSecret(pod.Namespace, secretRef.Name)
 		if err != nil {
 			klog.InfoS("Unable to retrieve pull secret, the image pull may not succeed.", "pod", klog.KObj(pod), "secret", klog.KObj(secret), "err", err)
+			failedPullSecrets = append(failedPullSecrets, secretRef.Name)
 			continue
 		}
 
 		pullSecrets = append(pullSecrets, *secret)
 	}
 
+	if len(failedPullSecrets) > 0 {
```

Comment:
One thing I notice is that you only display the secret name but other cases in this file also display both namespace and name.  

```
			if len(invalidKeys) > 0 {
				sort.Strings(invalidKeys)
				kl.recorder.Eventf(pod, v1.EventTypeWarning, "InvalidEnvironmentVariableNames", "Keys [%s] from the EnvFrom configMap %s/%s were skipped since they are considered invalid environment variable names.", strings.Join(invalidKeys, ", "), pod.Namespace, name)
			}
```

You could cause a confusing error message where a secret exists but not in the same namespace as the pod.

Reply:
I considered this but discarded it when I read this discussion between @SergeyKanzhelev  and @liggitt  https://github.com/kubernetes/kubernetes/pull/104535#discussion_r747254800. This PR is based on that PR, which got the `lgtm` label before it rot.

They don't think it necessary to add the namespace, because the events are already namespaced and the secrets can only be within that namespace.

Reply:
I see.  That sounds good.  

/lgtm